enterprise/internal/codeintel/shared/types/indexers.go

```diff
@@ -1,172 +1,138 @@
-		Name: "msft/lsif-node",
-		URN:  "github.com/Microsoft/lsif-node",
-	}
-	scipTypescript = CodeIntelIndexer{
-		Name: "scip-typescript",
-		URN:  "github.com/sourcegraph/scip-typescript",
-	}
-	scipJava = CodeIntelIndexer{
-		Name: "scip-java",
-		URN:  "github.com/sourcegraph/scip-java",
-	}
-	msftJava = CodeIntelIndexer{
-		Name: "msft/lsif-java",
-		URN:  "github.com/Microsoft/lsif-java",
-	}
-	lsifGo = CodeIntelIndexer{
-		Name: "lsif-go",
-		URN:  "github.com/sourcegraph/lsif-go",
-	}
-	lsifClang = CodeIntelIndexer{
-		Name: "lsif-clang",
-		URN:  "github.com/sourcegraph/lsif-clang",
-	}
-	lsifCPP = CodeIntelIndexer{
-		Name: "lsif-cpp",
-		URN:  "github.com/sourcegraph/lsif-cpp",
-	}
-	lsifDart = CodeIntelIndexer{
-		Name: "lsif-dart",
-		URN:  "github.com/sourcegraph/lsif-dart",
-	}
-	workivaDart = CodeIntelIndexer{
-		Name: "lsif_indexer",
-		URN:  "github.com/Workiva/lsif_indexer",
-	}
-	hieLSIF = CodeIntelIndexer{
-		Name: "hie-lsif",
-		URN:  "github.com/mpickering/hie-lsif",
-	}
-	lsifJsonnet = CodeIntelIndexer{
-		Name: "lsif-jsonnet",
-		URN:  "github.com/sourcegraph/lsif-jsonnet",
-	}
-	lsifOcaml = CodeIntelIndexer{
-		Name: "lsif-ocaml",
-		URN:  "github.com/rvantonder/lsif-ocaml",
-	}
-	scipPython = CodeIntelIndexer{
-		Name: "scip-python",
-		URN:  "github.com/sourcegraph/scip-python",
-	}
-	rustAnalyzer = CodeIntelIndexer{
-		Name: "rust-analyzer",
-		URN:  "github.com/rust-analyzer/rust-analyzer",
-	}
-	lsifPHP = CodeIntelIndexer{
-		Name: "lsif-php",
-		URN:  "github.com/davidrjenni/lsif-php",
-	}
-	lsifTerraform = CodeIntelIndexer{
-		Name: "lsif-terraform",
-		URN:  "github.com/juliosueiras/lsif-terraform",
-	}
-	lsifDotnet = CodeIntelIndexer{
-		Name: "lsif-dotnet",
-		URN:  "github.com/tcz717/LsifDotnet",
-	}
-	scipRuby = CodeIntelIndexer{
-		Name: "scip-ruby",
-		URN:  "github.com/sourcegraph/scip-ruby",
-	}
-)
-
+// AllIndexers is a list of all detectable/suggested indexers known to Sourcegraph.
+// Two indexers with the same language key will be preferred according to the given order.
 var AllIndexers = []CodeIntelIndexer{
-	lsifNode,
-	msftNode,
-	scipTypescript,
-	scipJava,
-	msftJava,
-	lsifGo,
-	lsifClang,
-	lsifCPP,
-	lsifDart,
-	workivaDart,
-	hieLSIF,
-	lsifJsonnet,
-	lsifOcaml,
-	scipPython,
-	rustAnalyzer,
-	lsifPHP,
-	lsifTerraform,
-	lsifDotnet,
-	scipRuby,
+	// C++
+	makeInternalIndexer("C++", "scip-clang"),
+	makeInternalIndexer("C++", "lsif-clang"),
+	makeInternalIndexer("C++", "lsif-cpp"),
+
+	// Dart
+	makeInternalIndexer("Dart", "lsif-dart"),
+	makeIndexer("Dart", "lsif_indexer", "github.com/Workiva/lsif_indexer"),
+
+	// DotNet
+	makeInternalIndexer("DotNet", "scip-dotnet"),
+	makeIndexer("DotNet", "lsif-dotnet", "github.com/tcz717/LsifDotnet"),
+
+	// Go
+	makeInternalIndexer("Go", "scip-go"),
+	makeInternalIndexer("Go", "lsif-go"),
+
+	// HIE
+	makeIndexer("HIE", "hie-lsif", " github.com/mpickering/hie-lsif"),
+
+	// Jsonnet
+	makeInternalIndexer("Jsonnet", "lsif-jsonnet"),
+
+	// JVMk
```

Comment:
Just to make grepping for these languages easier
```suggestion
	// JVM: Java, Scala, Kotlin
```